notebooks/sagemaker/Deploy command-r.ipynb

```diff
@@ -0,0 +1,500 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "message = \"Write a LinkedIn post about starting a career in tech:\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### C. Perform real-time inference"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response = co.chat(message=message, stream=False)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### D. Visualize output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(response)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### E. Streaming Chat"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "message = \"Write a LinkedIn post about starting a career in tech:\"\n",
+    "\n",
+    "response = co.chat(message=message, stream=True)\n",
+    "\n",
+    "# stream events back\n",
+    "for res in response:\n",
+    "    print(res)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### F. Chat with documents (RAG)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "message=\"How deep in the Mariana Trench\"\n",
+    "documents = [\n",
+    "    {\n",
+    "       \"id\": \"national_geographic_everest\",\n",
+    "       \"title\": \"Height of Mount Everest\",\n",
+    "       \"snippet\": \"The height of Mount Everest is 29,035 feet\",\n",
+    "       \"url\": \"https://education.nationalgeographic.org/resource/mount-everest/\",\n",
+    "    },\n",
+    "    {\n",
+    "        \"id\": \"national_geographic_mariana\",\n",
+    "        \"title\": \"Depth of the Mariana Trench\",\n",
+    "        \"snippet\": \"The depth of the Mariana Trench is 36,070 feet\",\n",
+    "        \"url\": \"https://www.nationalgeographic.org/activity/mariana-trench-deepest-place-earth\",\n",
+    "    }\n",
+    "]\n",
+    "\n",
+    "response = co.generate(prompt=message, documents=documents, stream=False)\n",
+    "\n",
+    "print(response)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### G. Generate search queries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "message=\"What is the height of Mount Everest?\"\n",
+    "\n",
+    "response = co.chat(message=message, search_queries_only=True, stream=False)\n",
+    "\n",
+    "print(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### H. Tool inputs"
```

Comment:
Seems kinda weird to have this particular section since the tool doesn't have any results to use. Maybe we can get rid of it and keep the Tool Results part only?

Reply:
Tools are used in 3 steps, getting tool inputs + calling the tool and then generating with tool results